erpnext/manufacturing/doctype/work_order/work_order.py

```diff
@@ -436,6 +436,18 @@ def get_items_and_operations_from_bom(self):
 		self.set_work_order_operations()
 
 		return check_if_scrap_warehouse_mandatory(self.bom_no)
+	
+	def get_item_bom_from_barcode(self,barcode):
```

Comment:
Make this function a whitelisted function
There is no need to pass the work order doc since it it not used anywhere in this function